File: app/src/main/java/br/com/smartfit/omdbapp/ui/detalhesitem/DetalhesItemContrato.java
package br.com.smartfit.omdbapp.ui.detalhesitem;

import br.com.smartfit.omdbapp.model.Item;

public interface DetalhesItemContrato {

    interface View{
        void inicializar();
        void mostrarDados(Item item);
        void mostrarMensagemErro(int mensagem);
        void esconderPbItem();
    }

    interface Presenter{
        void buscarDados(String id);
    }
}

File: app/src/main/java/br/com/smartfit/omdbapp/ui/detalhesitem/DetalhesItemContrato.kt
package br.com.smartfit.omdbapp.ui.detalhesitem

import br.com.smartfit.omdbapp.model.Item

interface DetalhesItemContrato {

    interface View {
        fun inicializar()
        fun mostrarDados(item: Item)
        fun mostrarMensagemErro(mensagem: Int)
        fun esconderPbItem()
    }

    interface Presenter {
        fun buscarDados(id: String?)
    }
}